source/slang/slang.cpp

```diff
@@ -4231,6 +4421,26 @@ void RenamedEntryPointComponentType::acceptVisitor(
         this, as<EntryPoint::EntryPointSpecializationInfo>(specializationInfo));
 }
 
+SlangResult RenamedEntryPointComponentType::computeDependencyBasedHash(slang::Checksum* outHashCode)
+{
+    slang::Checksum hashCode;
+    MD5HashGen hashGen;
+    MD5Context context;
+    hashGen.init(&context);
+
+    auto nameOverride = getEntryPointNameOverride(0);
+    hashGen.update(&context, nameOverride);
+
+    slang::Checksum baseHash;
+    getBase()->computeDependencyBasedHash(&baseHash);
+    hashGen.update(&context, baseHash);
+
+    hashGen.finalize(&context, &hashCode);
+
+    *outHashCode = hashCode;
+    return SLANG_OK;
+}
```

Comment:
My sense is also that we shouldn't bother implementing this logic around `EntryPoint` or `RenamedEntryPointComponentType`. This is again a case where a single non-recursive routine written for any `ComponentType` can iterate over the entry points of that component (without needing to know which sub-type it is), and ask for their names (whether or not they've been renamed). That should result in simpler and more general code.

Reply:
Logic to handle entry point name and name override as specified by `entryPointIndex` moved up to `CompositeComponentType`. Both `EntryPoint` and `RenamedEntryPointComponentType` immediately return now.

Reply:
They should be moved into ComponentType class instead of CompositeComponentType. See the other comments on how to restructure these things.

Reply:
Fixed locally, will be in next push